Packs/ApiModules/Scripts/MicrosoftApiModule/MicrosoftApiModule.py

```diff
@@ -975,11 +978,17 @@ def _oproxy_authorize(self, resource: str = '', scope: str | None = None) -> tup
         """
         content = self.refresh_token or self.tenant_id
         headers = self._add_info_headers()
+
+        should_delay_request()
         oproxy_response = self._oproxy_authorize_build_request(headers, content, scope, resource)
 
         if not oproxy_response.ok:
+            next_request_time = calculate_next_request_time()
+            set_retry_mechanism_arguments(next_request_time=next_request_time)
             self._raise_authentication_error(oproxy_response)
 
+        # In case of success, reset the retry mechanism arguments.
+        set_retry_mechanism_arguments()
```

Comment:
let's do this in a single call to `get_integration_context()` and `set_integration_context()` (each). the reason is to reduce the number of interactions with the server.
also consider using a decorator